src/main/scala/lambdaone/toolbox/EventStore.scala

```diff
@@ -0,0 +1,60 @@
+package lambdaone.toolbox
+
+import cats.data.NonEmptyList
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+
+trait EventStore[F[_], A] {
+
+  def append(a: A): F[A]
+
+  def latest: F[A]
+
+  def loadAll: F[NonEmptyList[A]]
+
+  // Scan from earliest to latest
+  def find(f: A => Boolean): F[Option[A]]
+
+  def findMap[B](pf: PartialFunction[A, B]): F[Option[B]]
+
+  def filterMap[B](pf: PartialFunction[A, B]): F[List[B]]
+}
+
+object EventStore {
+
+  def inMem[F[_], A](first: A)(implicit F: Sync[F]): F[EventStore[F, A]] =
```

Comment:
This is an implementation given a first event A, we can also do a pure implementation using the `State` monad, and also this implementation can be still kept as a typeclass instance by doing in with `Kleisli`